Legend rework and Geography filter removal

brett-stephen opened the pull request:
## Briefly describe the feature or bug that this PR addresses.
- Adds toggles to each individual pin type in the legend
- Removes estimate grade from filter
## Please link the Airtable ticket associated with this PR.
https://airtable.com/tbli2lWQHAqBa6ZcI/viw8Ro4zYPYcBGXRw/recHMdR15tgClYpiG?blocks=hide

## If applicable, include screenshots of the feature/bugfix introduced by this PR (Please include both desktop and mobile if there is a significant UI change).
![image](https://user-images.githubusercontent.com/22464147/112777715-262fd200-9000-11eb-8b6f-64daac796354.png)
## Describe the steps you took to test the feature/bugfix introduced by this PR.
- Some quick smoke testing to verify the filters work correctly

## Does this PR depend on any recent backend work? If so, please include the PR number from the iit-backend repo and associated Airtable ticket link.
- may need to deprecate the filters for estimate grade

atatmaja commented:
Code itself looks good, but are we sure it's better to do use a "estimate grade" layer instead of just making the legend toggles work like filters? I guess the pros and cons of this approach that I have in mind are:

- Pros: May achieve faster performance on toggle 
- Cons: Less code reuse / more to maintain, if intended behaviour around the estimate grade filter/toggle ever changes we would have to change in multiple places instead of just making a change on the backend

atatmaja commented on src/context.tsx:
```diff
@@ -164,9 +190,5 @@ const reducer = (state: State, action: Record<string, any>): State => {
-      {props.children}
-    </AppContext.Provider>
-  );
+  return <AppContext.Provider value={[state, dispatch]}>{props.children}</AppContext.Provider>;
```
did your linter do this? looks kinda ugly tbh lmao

atatmaja commented on src/components/map/Legend.tsx:
```diff
@@ -9,40 +9,50 @@ import { MapSymbology } from "components/map/MapConfig"
         <div className="legend-item">
           <i className="block" style={{ background: MapSymbology.CountryFeature.HasData.Color, outlineWidth: 1, outlineStyle: "solid" }}></i>
-          <p className={isMobileDeviceOrTablet ? "mobile-text px-2" : "px-1"}>{Translate("CountryEstimatesExist")}</p>
+          <p>{Translate("CountryEstimatesExist")}</p>
```
if we're no longer applying conditional styling on mobile vs desktop, did we test to see how it looks on both screen sizes?

brett-stephen replied:
Legend is compact enough where we don't need to change styling for mobile. This version takes less space and looks better than the prev mobile legend
> 
> 
> ![image](https://user-images.githubusercontent.com/22464147/113496413-90d88600-94b6-11eb-850f-4df633b5c591.png)
> 
> Updated Legend to be more compact & use proper semantic ui check boxes

brett-stephen commented:
![image](https://user-images.githubusercontent.com/22464147/113496413-90d88600-94b6-11eb-850f-4df633b5c591.png)

Updated Legend to be more compact & use proper semantic ui check boxes

atatmaja commented:
Can u share screenshot of what it looks like on mobile?

atatmaja commented:
Not necessary for this PR but would be cool if we could make the legend collapsable (for power users who are familiar with the legend and want more screen real estate) @brett-stephen @jordanvanwyk what do you think

atatmaja reviewed (approved):
Code LGTM, can merge once we agree that mobile version looks ok

jordanvanwyk commented:
maybe we can fit legend into the side bar to fit with tableau as another option. we can do more mocks 
https://www.figma.com/file/duS85QuRG7WPz6AlPwEmBk/Legend?node-id=0%3A1 <https://www.figma.com/file/duS85QuRG7WPz6AlPwEmBk/Legend?node-id=0:1>


> On Apr 4, 2021, at 20:44, Austin Atmaja ***@***.***> wrote:
> 
> 
> Not necessary for this PR but would be cool if we could make the legend collapsable (for power users who are familiar with the legend and want more screen real estate) @brett-stephen <https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbrett-stephen&data=04%7C01%7C%7Ccd9403d2294445cafc7b08d8f7cc09e5%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637531802997282873%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=yx7ewWT2HSAtdWBi%2BlZlM119Nt3iKSswxUgVDJ6U9I0%3D&reserved=0> @jordanvanwyk <https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjordanvanwyk&data=04%7C01%7C%7Ccd9403d2294445cafc7b08d8f7cc09e5%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637531802997292867%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=vWn9SKecCOPpMeN4SgEV4PIwGXsPhuR2CkdYknrRSRA%3D&reserved=0> what do you think
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fserotracker%2Fsero-can-webapp%2Fpull%2F221%23issuecomment-813127504&data=04%7C01%7C%7Ccd9403d2294445cafc7b08d8f7cc09e5%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637531802997292867%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=DATZ9HYD35UJ1SVsWPsUMrY2IWaIKnTmr7gOuoEJcLU%3D&reserved=0>, or unsubscribe <https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAFLRZCDVUUBY7VBLL2W2RV3THEBYRANCNFSM4Z5VJFPQ&data=04%7C01%7C%7Ccd9403d2294445cafc7b08d8f7cc09e5%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637531802997302859%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=ZUUKC5uhEhTbGywTfDNjp97MJTeNlBo0HOCger9a0u8%3D&reserved=0>.
>

brett-stephen commented:
We could make the legend have the same behavior as filters and map statistics on mobile. I'm not sure if being able to hide the legend would be an issue for the WHO

brett-stephen commented:
Moved mobile legend into a sidebar button 
State - Non expanded:
![image](https://user-images.githubusercontent.com/22464147/113620561-b3ce7b80-9617-11eb-8bf5-1a02eaefeddd.png)
State - Expanded: 
![image](https://user-images.githubusercontent.com/22464147/113620613-cba5ff80-9617-11eb-9c28-739bd810aa84.png)

brett-stephen commented:
Removed legend toggle, and cleaned up & adjusted the CSS.
Desktop:
![image](https://user-images.githubusercontent.com/22464147/114108052-7aa04080-988f-11eb-8c77-83c447bd9425.png)

Mobile:
![image](https://user-images.githubusercontent.com/22464147/114108727-f2229f80-9890-11eb-8ed0-584a1ec68ce2.png)
